### PR TITLE
`mark-merge-commits-in-list` - Support new commit list style

### PR DESCRIPTION
**Pull request by kovsu**
close: #7003 

## Test URLs

- isCommitList: https://github.com/typed-ember/ember-cli-typescript/commits/master?after=5ff0c078a4274aeccaf83382c0d6b46323f57397+174

### CODE REVIEW

**Comment by kovsu** [comment]
after:
image

**Comment by fregante** [comment]
Thank you! Can you also look at `dim-bots`?

- https://github.com/refined-github/refined-github/issues/7068

**Review by fregante — CHANGES_REQUESTED** [review]
So I think here (and maybe in the other PR) it can be greatly simplified. Most likely you don't need a `isLegacy` check:

- query `'.markdown-title a, a.markdown-title'` in the same function. The right element will be selected in both cases.
- add `.listviewitem` to the `$$()` array

I think that's about it, here. In the other feature `isLegacy` is still needed for the `flex-wrap` addition I think

This change is particularly necessary because the old view isn't really legacy, I just noticed that `isPRCommits` still uses it: https://github.com/refined-github/refined-github/pull/7087/commits

File: source/features/mark-merge-commits-in-list.tsx [hunk 1]
```diff
@@ -38,21 +38,37 @@ export function getCommitHash(commit: HTMLElement): string {
 	return $('a.markdown-title', commit)!.pathname.split('/').pop()!;
```
Comment (fregante) [inline source/features/mark-merge-commits-in-list.tsx:38]:
```suggestion
	return $([
		'a.markdown-title', // Old view style (before November 2023)
		'.markdown-title a',
	], commit)!.pathname.split('/').pop()!;
```

You can use array to comment on each selector

**Comment by kovsu** [comment]
> So I think here ~(and maybe in the other PR)~ it can be greatly simplified. Most likely you don't need a `isLegacy` check:
> 
> * query `'.markdown-title a, a.markdown-title'` in the same function. The right element will be selected in both cases.
> * add `.listviewitem` to the `$$()` array
> 
> I think that's about it, here. In the other feature `isLegacy` is still needed for the `flex-wrap` addition I think
> 
> This change is particularly necessary because the old view isn't really legacy, I just noticed that `isPRCommits` still uses it: [#7087 (commits)](https://github.com/refined-github/refined-github/pull/7087/commits)

I haven't used `select-dom` before. I will check it.

**Review by fregante — APPROVED** [review]

**Comment by kovsu** [comment]
Should I update the other PR?

**Comment by fregante** [comment]
Already done 😃

**Comment by fregante** [comment]
## Updated screenshot

Screenshot 3

**Comment by fregante** [comment]
If you'd like, it'd be good to fix the TODO in this file. You can see how it's used in other features

**Comment by kovsu** [comment]
like this? @fregante
image

**Comment by fregante** [comment]
Yes I think so ☺️ 

You can also drop the `duplicate` key

**Pull request merged**